src/swbf_fnv_hashes.cpp

```diff
@@ -1,2834 +1,3935 @@
-   "WaverTurn"_fnvp,
-   "weapon"_fnvp,
-   "Weapon"_fnvp,
-   "WeaponAmmo1"_fnvp,
-   "WeaponAmmo2"_fnvp,
-   "WeaponAmmo3"_fnvp,
-   "WeaponAmmo4"_fnvp,
-   "WeaponAmmo"_fnvp,
-   "WeaponChange"_fnvp,
-   "WeaponChangeSound"_fnvp,
-   "WeaponChannel1"_fnvp,
-   "WeaponChannel2"_fnvp,
-   "WeaponChannel3"_fnvp,
-   "WeaponChannel4"_fnvp,
-   "WeaponChannel"_fnvp,
-   "WeaponName1"_fnvp,
-   "WeaponName2"_fnvp,
-   "WeaponName3"_fnvp,
-   "WeaponName4"_fnvp,
-   "WeaponName"_fnvp,
-   "WEAPONSECTION"_fnvp,
-   "WeaponSection"_fnvp,
-   "WeaponShareAmmo1"_fnvp,
-   "WeaponShareAmmo2"_fnvp,
-   "WeaponShareAmmo3"_fnvp,
-   "WeaponShareAmmo4"_fnvp,
-   "WeaponShareAmmo"_fnvp,
-   "WeaponShareHeat1"_fnvp,
-   "WeaponShareHeat2"_fnvp,
-   "WeaponShareHeat3"_fnvp,
-   "WeaponShareHeat4"_fnvp,
-   "WeaponShareHeat"_fnvp,
-   "WeaponUnableSelect"_fnvp,
-   "WheelOmegaToU"_fnvp,
-   "WheelOmegaToV"_fnvp,
-   "WheelSection"_fnvp,
-   "WheelTexture"_fnvp,
-   "WheelVelocToU"_fnvp,
-   "WheelVelocToV"_fnvp,
-   "Widescreen"_fnvp,
-   "Width"_fnvp,
-   "WiggleAmount"_fnvp,
-   "WiggleSpeed"_fnvp,
-   "win"_fnvp,
-   "winddirection"_fnvp,
-   "WindDownSound"_fnvp,
-   "WindSound"_fnvp,
-   "windspeed"_fnvp,
-   "WindUpSound"_fnvp,
-   "WingModel"_fnvp,
-   "word"_fnvp,
-   "word_reformat"_fnvp,
-   "WorldAligned"_fnvp,
-   "WorldBrightness"_fnvp,
-   "WorldContrast"_fnvp,
-   "WorldFogRange"_fnvp,
-   "WorldHeight"_fnvp,
-   "WorldRange"_fnvp,
-   "WorldSaturation"_fnvp,
-   "WrapInput"_fnvp,
-   "YawLimits"_fnvp,
-   "YawSpread"_fnvp,
-   "YawTurnFactor"_fnvp,
-   "YOffset"_fnvp,
-   "ZeroGravity"_fnvp,
-   "ZoomFirstPerson"_fnvp,
-   "ZoomMax"_fnvp,
-   "ZoomMin"_fnvp,
-   "ZoomRate"_fnvp,
-   "ZoomTurnDivisorMax"_fnvp,
-   "ZoomTurnDivisorMin"_fnvp,
-   "ZOrder"_fnvp};
-}
+std::mutex dictionary_mutex;
 
 std::string lookup_fnv_hash(const std::uint32_t hash)
 {
+   std::string retVal;
+
+   dictionary_mutex.lock();
    const auto result = swbf_hashes.find(hash);
 
-   if (result != std::cend(swbf_hashes)) return std::string{result->second};
+   if (result != std::cend(swbf_hashes)) {
+      // this can be interesting when building a dict
+      //synced_cout::print("looked_up:", result->second, "\n");
+      retVal = std::string{result->second};
+   }
+   dictionary_mutex.unlock();
+
+   if (retVal.empty()) {
+      // here hex is nice for 2 reasons:
+      //   1. easier to see in hex editor
+      //   2. it matches the output of the ToolsFl\bin\hash.exe program
+      char hex_string[20];
+      sprintf(hex_string, "0x%x", hash);
+      synced_cout::print("Warning: Unknown hash looked up.\n"s, "   value: "s, hex_string,
+                         '\n');
+      retVal = std::string(hex_string);
+   }
+   return retVal;
+}
+
+// Read in a file of strings that can be resolved by the hash lookup
+void read_dictionary(const char* fileName)
+{
+   synced_cout::print("Reading dictionary: "s, fileName, '\n');
+   std::ifstream input(fileName);
```

Comment:
It is never checked if opening the file succeeded. Sadly because IO is one of the C++'s standard library's weaker points we have to do this ourselves by calling [std::ifstream::is_open](https://en.cppreference.com/w/cpp/io/basic_ifstream/is_open).

There is a real style nitpick as well for the most part but I feel this is too dense. Some line breaks help break up the function and make it easier to read (in my opinion, you may very reasonably disagree of course it is subjective after all.)
```suggestion
   synced_cout::print("Reading dictionary: "s, file_path, '\n');

   std::ifstream input(file_path);

   if (!input.is_open()) {
      throw std::runtime_error{"Failed to open file"s};
   }

```